### PR TITLE
Document remainingPath behaviour in README

### PR DESCRIPTION
**Pull request by aknuds1**
In README, document returned `remainingPath` property from `resolveData`.

Also clean up corresponding tests, making them define more properly what gets returned by `resolveData` (including `remainingPath`).

It solves [this](https://trello.com/c/AD43jA8t/45-readers-of-the-tupelojs-readme-should-understand-what-the-remaining-path-returned-from-the-resolve-request-means) card.

### CODE REVIEW

File: docs/file/lib/tupelo.js.html [hunk 1]
```diff
@@ -340,10 +347,14 @@
    */
 
   /**
-   * Resolve data from the user portion of the chain tree
+   * Resolve data from the user portion of the chain tree.
```
Comment (tobowers) [inline docs/file/lib/tupelo.js.html]: maybe This should be:
`
Resolve data from the user portion ("/tree/data") of the chain tree.
`

Comment (aknuds1) [reply docs/file/lib/tupelo.js.html]: @tobowers Makes sense.

**Comment by aknuds1** [comment]
Checking why tests started failing.

**Comment by aknuds1** [comment]
@tobowers Are you able to review this? I fixed the test failures.

**Pull request merged**